src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.cs

```diff
@@ -277,20 +298,9 @@ void ProbeDirectory(string manifestDirectory, string featureBand)
                 }
             }
 
-            //  Load manifests from workload set, if any.  This will override any manifests for the same IDs that were loaded previously in this method
-            if (_workloadSet != null)
+            if (!_workloadSetDominatesInstallState)
```

Comment:
I don't think we need to change the order of when we load the manifests here.  Rather, if a workload set is specified in global.json, we should ignore the install state entirely.  So you could undo the refactoring of `LoadManifestsFromWorkloadSet` into a separate method, and skip the logic that processes `_manifestsFromInstallState` if a workload set was specified via global.json (and probably also if it was specified by being passed to the constructor).

Reply:
Those are both included from the block before the table (lines 82-90)

Reply:
This is a question of when we know what.

When we're trying to generate _workloadSet, we have all the relevant information at our fingertips, most importantly where it came from. At this point, though, we just know that there is a workload set; it might've come from a global.json (in which case it should be prioritized over an install state file), or it might've just been the highest numbered workload set we could find, in which case it should be prioritized lower. This _workloadSetDominatesInstallState field is intended to disambiguate that.

Reply:
We can still keep a boolean to let us know what we should do.  But there's no reason to load the manifests from the install state and then the manifests from the workload set.  We should load the manifests from the workload set, and then either load the manifests from the install state, or skip it.

I think a name like `_useManifestsFromInstallState` would be clearer in that case.  So we'd have something like this:

```c#
LoadManifestsFromWorkloadSet();

if (_manifestsFromInstallState != null && _useManifestsFromInstallState)
{
    //  Code here to load manifests from the install state
}
```

Then we only have one call to `LoadManifestsFromWorkloadSet`, so we can remove that as a separate method and put the implementation back into `GetManifests` directly.

`_useManifestsFromInstallState` should be set to false if we get a workload version from global.json, or if the workload set version was specified via the constructor.  Otherwise it should be set to true.

Does that make sense?